src/main/java/org/apache/xmlbeans/XmlOptions.java

```diff
@@ -1070,6 +1088,37 @@ public boolean isCompileDownloadUrls() {
         return hasOption(XmlOptionsKeys.COMPILE_DOWNLOAD_URLS);
     }
 
+    /**
+     * If this option is set, then the schema compiler will use utf_8 to generate java source file
+     *
+     */
+    public XmlOptions setCompileSourceCodeEncoding () {
+        return setCompileSourceCodeEncoding (true);
+    }
+
+    public XmlOptions setCompileSourceCodeEncoding (boolean b) {
```

Comment:
my suggestion was that the input should be a string - the name of an encoding, eg `"utf-8"`

Reply:
Okay, I'll fix it tomorrow

Reply:
I've created #16 - this is just an initial change, needs more work

Reply:
> I've created #16 - this is just an initial change, needs more work

Maybe you have the wrong branch? My changes are based on branch REL_5_1_1 instead of trunk, but your PR#16 is based on trunk

Reply:
I updated the code using all your changes and added the ones I missed before, is there anything else I need to modify？

Reply:
Testing these changes is hard. XMLBeans does't have great support for testing the code generation support.
If you can build this yourself and see if it does the job for you then that's great.
I will try to get #16 merged at some stage - but only when I have time to build and test it.
I am omitting some of your changes (like extra namespaces) because I would prefer to do separate PRs.

Reply:
My bad, I will upload it to a separate PR soon,and for the version https://github.com/apache/xmlbeans/pull/16 I will test it tomorrow when I go to work👍

Reply:
@pjfanning I just fixed a small problem and then used the jar with the command line parameters "-sourcecodeencoding UTF-8 -usejavashortname" to test my project. The build was successful and the unit tests passed completely. You should be fine after you apply my commit.（：

Reply:
I spotted that issue independently and already have a fix in #16